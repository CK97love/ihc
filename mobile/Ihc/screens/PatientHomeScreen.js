import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  TouchableOpacity,
  Text,
  View,
} from 'react-native';
import { Col, Row, Grid } from "react-native-easy-grid";

export default class PatientHomeScreen extends Component<{}> {
  /*
   * Expects:
   *  {
   *    name: string, patient's name (for convenience)
   *    patientKey: string
   *  }
   */
  constructor(props) {
    super(props);
  }

  // TODO all of these except medications
  goToTriage = () => {
    this.props.navigator.push({
      screen: 'Ihc.PatientSelectScreen',
      title: 'Back to patient'
    });
  }

  goToSoap = () => {
    this.props.navigator.push({
      screen: 'Ihc.SoapScreen',
      title: 'Back to patient',
      passProps: { name: this.props.name, patientKey: this.props.patientKey }
    });
  }

  goToMedicationList = () => {
    this.props.navigator.push({
      screen: 'Ihc.MedicationScreen',
      title: 'Back to patient',
      passProps: { name: this.props.name, patientKey: this.props.patientKey }
    });
  }

  goToHistory = () => {
    this.props.navigator.push({
      screen: 'Ihc.PatientSelectScreen',
      title: 'Back to patient',
    });
  }

  goToGrowthChart = () => {
    this.props.navigator.push({
      screen: 'Ihc.GrowthChartScreen',
      title: 'Back to patient',
      passProps: { patientKey: this.props.patientKey}
    });
  }

  render() {
    const date = new Date();
    //const dateString = `${date.getMonth()} ${date.getDate()}, ${date.getYear()}`;
    const dateString = date.toDateString();
    return (
      <View style={styles.container}>
        <Text style={styles.title}>
          {this.props.name}
        </Text>

        <Text style={styles.title}>
          {dateString}
        </Text>

        <View style={styles.gridContainer}>
          <Grid>
            <Col style={styles.col}>
              <TouchableOpacity style={styles.buttonContainer}
                  onPress={this.goToTriage}>
                <Text style={styles.button}>Triage</Text>
              </TouchableOpacity>

              <TouchableOpacity style={styles.buttonContainer}
                  onPress={this.goToSoap}>
                <Text style={styles.button}>SOAP</Text>
              </TouchableOpacity>

              <TouchableOpacity style={styles.buttonContainer}
                  onPress={this.goToMedicationList}>
                <Text style={styles.button}>Medications</Text>
              </TouchableOpacity>
            </Col>

            <Col style={styles.col}>
              <TouchableOpacity style={styles.buttonContainer}
                  onPress={this.goToHistory}>
                <Text style={styles.button}>History</Text>
              </TouchableOpacity>

              <TouchableOpacity style={styles.buttonContainer}
                  onPress={this.goToGrowthChart}>
                <Text style={styles.button}>Growth Chart</Text>
              </TouchableOpacity>
            </Col>
          </Grid>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  gridContainer: {
    flex: 1,
    maxWidth: '80%',
    alignItems: 'center',
  },
  col: {
    alignItems: 'center',
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 5,
  },
  buttonContainer: {
    width: 150,
    margin: 10,
    padding: 8,
    elevation: 4,
    borderRadius: 2,
    backgroundColor: '#2196F3',
  },
  button: {
    fontWeight: '500',
    color: '#fefefe',
    textAlign: 'center',
  }
});
