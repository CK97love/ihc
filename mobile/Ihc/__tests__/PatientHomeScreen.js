import 'react-native';
import React from 'react';
import PatientHomeScreen from '../screens/PatientHomeScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const json = renderer.create(
    <PatientHomeScreen />
  ).toJSON();
  expect(json).toMatchSnapshot();
});
