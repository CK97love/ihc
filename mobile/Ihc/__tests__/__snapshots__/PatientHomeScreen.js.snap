// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 0,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "margin": 5,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 20,
            "margin": 5,
            "textAlign": "center",
          }
        }
      >
        Tue Apr 10 2018
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "maxWidth": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2196F3",
                  "borderRadius": 2,
                  "elevation": 4,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 4,
                  "opacity": 1,
                  "padding": 8,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fefefe",
                      "fontWeight": "500",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Triage
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2196F3",
                  "borderRadius": 2,
                  "elevation": 4,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 4,
                  "opacity": 1,
                  "padding": 8,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fefefe",
                      "fontWeight": "500",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                SOAP
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2196F3",
                  "borderRadius": 2,
                  "elevation": 4,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 4,
                  "opacity": 1,
                  "padding": 8,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fefefe",
                      "fontWeight": "500",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Medications
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2196F3",
                  "borderRadius": 2,
                  "elevation": 4,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 4,
                  "opacity": 1,
                  "padding": 8,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fefefe",
                      "fontWeight": "500",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                History
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2196F3",
                  "borderRadius": 2,
                  "elevation": 4,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 4,
                  "opacity": 1,
                  "padding": 8,
                  "width": "80%",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#fefefe",
                      "fontWeight": "500",
                      "textAlign": "center",
                    },
                    undefined,
                  ]
                }
              >
                Growth Chart
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "textAlign": "center",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "green",
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
